SwanCustomEnvironments/swancustomenvironments/scripts/builders/conda.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Create the environment and install packages
+${BUILDER_VERSION} create -p ${ENV_PATH} --file ${REQ_PATH} -y | tee -a ${LOG_PATH}
+
+# Activate the environment
+_log "Setting up the environment..."
+ACTIVATE_ENV_CMD="source activate ${ENV_PATH}"
```

Comment:
What did you find in the end about this?

From what I can read, `source activate` was the way to activate environments in older versions of conda. It is now not a documented way in the conda docs:

https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#activating-an-environment

Did you find why our conda does not have the activate option? Is it just a matter of upgrading our conda in the user image? That could be done.

Reply:
From what I searched, I believe with these new changes it may work. I still gotta test it some more times but it looks ok from what I got. It could not find `conda activate` because we needed to `source /opt/conda/etc/profile.d/conda.sh` in order to properly recognize conda commands, kinda similar to `acc-py` in the sense that we first need to activate acc-py and then create the environment.

Reply:
What are the effects of running `source /opt/conda/etc/profile.d/conda.sh`? Does it have any implication for the environment?

Reply:
It wouldn't have negative implications I believe, since all it does is defining conda related variables and functions. Meanwhile, yesterday evening, I found out that we can make it even simpler. Instead of sourcing `/opt/conda/etc/profile.d/conda.sh`, we can source `/home/$USER/.bashrc`. It changes a couple variables, including `$PATH`. I still need to fully understand why and how this is working.

Reply:
What else is in .bashrc? Any possible side effect?

Reply:
I believe not, this is what mine has
```
# .bashrc

# Source global definitions
if [ -f /etc/bashrc ]; then
        . /etc/bashrc
fi

# User specific environment
if ! [[ "$PATH" =~ "$HOME/.local/bin:$HOME/bin:" ]]
then
    PATH="$HOME/.local/bin:$HOME/bin:$PATH"
fi
export PATH

# Uncomment the following line if you don't like systemctl's auto-paging feature:
# export SYSTEMD_PAGER=

# User specific aliases and functions
if [ -d ~/.bashrc.d ]; then
        for rc in ~/.bashrc.d/*; do
                if [ -f "$rc" ]; then
                        . "$rc"
                fi
        done
fi

unset rc
eval "$(command conda shell.bash hook 2> /dev/null)"
```

Reply:
We got the scripts working fine. We're activating the environment using conda with no warning messages being displayed

Reply:
So iiuc the piece of code that enables conda commands is:
```
eval "$(command conda shell.bash hook 2> /dev/null)"
```
Why not just doing that before conda activate?